future/interactions.go

```diff
@@ -0,0 +1,114 @@
+package future
+
+import (
+	"crypto/sha512"
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/algorand/go-algorand-sdk/abi"
+)
+
+type Arg struct {
+	Name    string `json:"name"`
+	AbiType string `json:"type"`
+	Desc    string `json:"desc,omitempty"`
+}
+
+type Return struct {
+	AbiType string `json:"type"`
+	Desc    string `json:"desc,omitempty"`
+}
+
+type Method struct {
+	Name    string `json:"name"`
+	Desc    string `json:"desc,omitempty"`
+	Args    []Arg  `json:"args"`
+	Returns Return `json:"returns"`
+}
+
+func MethodFromSignature(methodStr string) (Method, error) {
+	openCnt := strings.Count(methodStr, "(")
+	if openCnt == 0 || openCnt > 1 {
+		return Method{}, errors.New("Method signature has invalid format")
+	}
+
+	closeCnt := strings.Count(methodStr, ")")
+	if closeCnt == 0 || closeCnt > 1 {
+		return Method{}, errors.New("Method signature has invalid format")
+	}
+
+	openIdx := strings.Index(methodStr, "(")
+	closeIdx := strings.Index(methodStr, ")")
+	if openIdx > closeIdx {
+		return Method{}, errors.New("Method signature has invalid format")
+	}
+
+	// signature must include method name
+	if openIdx == 0 {
+		return Method{}, errors.New("Method signature has invalid format")
+	}
+
+	name := methodStr[:openIdx]
+	stringArgs := strings.Split(methodStr[openIdx+1:closeIdx], ",")
+	returnType := methodStr[closeIdx+1:]
```

Comment:
It seems to be that it does not allow tuple value existing in method signature?

Reply:
Good catch, fixed